flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/exceptionhistory/RootExceptionHistoryEntry.java

```diff
@@ -107,26 +121,33 @@ public static RootExceptionHistoryEntry fromExceptionHistoryEntry(
     public static RootExceptionHistoryEntry fromGlobalFailure(ErrorInfo errorInfo) {
         Preconditions.checkNotNull(errorInfo, "errorInfo");
         return fromGlobalFailure(
-                errorInfo.getException(), errorInfo.getTimestamp(), Collections.emptyList());
+                errorInfo.getException(),
+                errorInfo.getTimestamp(),
+                FailureEnricherUtils.EMPTY_FAILURE_LABELS,
+                Collections.emptyList());
     }
 
     private static RootExceptionHistoryEntry createRootExceptionHistoryEntry(
             Throwable cause,
             long timestamp,
+            CompletableFuture<Map<String, String>> failureLabels,
             @Nullable String failingTaskName,
             @Nullable TaskManagerLocation taskManagerLocation,
             Iterable<Execution> executions) {
         return new RootExceptionHistoryEntry(
                 cause,
                 timestamp,
+                failureLabels,
                 failingTaskName,
                 taskManagerLocation,
                 StreamSupport.stream(executions.spliterator(), false)
                         .filter(execution -> execution.getFailureInfo().isPresent())
                         .map(
                                 execution ->
                                         ExceptionHistoryEntry.create(
-                                                execution, execution.getVertexWithAttempt()))
+                                                execution,
+                                                execution.getVertexWithAttempt(),
+                                                failureLabels))
```

Comment:
These concurrent failures can be quite different from the root one. Assigning the same failure labels to them may be confusing. e.g. labeled a concurrent Flink system error to be `USER` when the root failure is a `USER` error.
Given that the concurrent failures are not failure root causes, I think they are not that critical. Therefore I think it's acceptable to assign empty labels to them.

Reply:
Thanks both! Sure, changed to `EMPTY_FAILURE_LABELS`

Labels are currently part of `ExceptionHistoryEntry` -- lets think about moving them  upas part of the Rest Api ticket.